core/src/test/scala/unit/kafka/log/AbstractLogCleanerIntegrationTest.scala

```diff
@@ -118,4 +121,30 @@ abstract class AbstractLogCleanerIntegrationTest {
       logDirFailureChannel = new LogDirFailureChannel(1),
       time = time)
   }
+
+  def codec: CompressionType
+  def counter: Int
+  def incCounter: Int
```

Comment:
For counter and incCounter, the implementations in the callers seem to be the same. Could we consolidate them here?